src/jdk.crypto.cryptoki/share/native/libj2pkcs11/p11_crypt.c

```diff
@@ -166,15 +169,13 @@ Java_sun_security_pkcs11_wrapper_PKCS11_C_1Encrypt
                                     (CK_BYTE_PTR)(outBufP + jOutOfs),
                                     &ckEncryptedLen);
 
-    ckAssertReturnValueOK(env, rv);
-
-cleanup:
     if (directIn == 0 && inBufP != NULL) {
```

Comment:
with this change, inBufP and outBufP should be non-null when directIn ==0 and directOut == 0, right?

Reply:
Same goes for the rest of the file. Otherwise, changes look good.

Reply:
Good point, will remove the extra checks.